buffer/src/test/java/io/netty5/buffer/tests/BufferCompositionTest.java

```diff
@@ -213,6 +213,37 @@ public void extendingCompositeBufferMustIncreaseCapacityByGivenBuffer() {
         }
     }
 
+    @Test
+    public void extendingCompositeBufferWithUnwrittenSpanMadeReadOnlyMustIncreaseCapacityByGivenBuffer() {
+        try (BufferAllocator allocator = BufferAllocator.onHeapUnpooled();
+             CompositeBuffer composite = allocator.compose().ensureWritable(255);
+             Buffer buffer = BufferAllocator.onHeapUnpooled().allocate(8).implicitCapacityLimit(8)) {
+
+            composite.writeLong(0x0102030405060708L);
+            composite.makeReadOnly();
+            assertEquals(255, composite.capacity());
+
+            buffer.writeLong(0x0807060504030201L);
+            buffer.makeReadOnly();
+            assertEquals(8, buffer.capacity());
+
+            composite.extendWith(buffer.send());
+
+            assertThat(composite.writerOffset()).isEqualTo(16);
+            assertThat(composite.readerOffset()).isZero();
+            assertThat(composite.readableBytes()).isEqualTo(16);
+
+            final byte[] copyOfComposite = new byte[16];
+
+            composite.readBytes(copyOfComposite, 0, copyOfComposite.length);
+
+            final Buffer copiedBuffer = BufferAllocator.offHeapUnpooled().copyOf(copyOfComposite);
```

Comment:
This copied buffer isn't closed.

Why not read the expected long values directly out of the composite buffer?

Reply:
> This copied buffer isn't closed.

Fixed that.

> 
> Why not read the expected long values directly out of the composite buffer?

Because the original issue is that the copy fails entirely I tend to want to check that the fix not only allows the method to not fall over but also produces the results I was expecting.